Modules/Core/Common/include/itkObject.h

```diff
@@ -156,6 +156,8 @@ class ITKCommon_EXPORT Object : public LightObject
 
   unsigned long
   AddObserver(const EventObject & event, Command *) const;
+  unsigned long
+  AddObserver(const EventObject & event, std::function<void(const EventObject &)> function) const;
```

Comment:
Do we need the non-const version?

Reply:
That was part of the reason, I removed the object argument to the callback, I was also considering removing the EventObject argument too.